### PR TITLE
UI-1 Get working as module

### PR DESCRIPTION
**Pull request by julianjuko**
Description coming soon

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { AdminLogo } from './AdminLogo'
+import { SideNav } from './SideNav'
+import { Navbar } from './Navbar'
+import { Footer } from './Footer'
+
+interface INavLink {
+  icon: any
```
Comment (lmiller1990) [inline src/index.ts]: should this be here? maybe it should be in the Narbar components?

File: src/style/colors.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export const PANELBLUE = 'rgb(81, 88, 115)'
+export const TEAL = 'rgb(0, 171, 176)'
+export const DARKTEAL = 'rgb(4, 156, 165)'
+export const LIGHTGREY = 'rgb(222, 226, 230)'
+export const DARKGREY = 'rgb(107, 107, 107)'
+export const DARKBLUE = 'rgb(0, 21, 61)'
+export const ADE = 'rgb(226, 175, 75)'
```
Comment (lmiller1990) [inline src/style/colors.ts]: I think ADE/SUB are specific to Cohere - not sure we need them here

File: package.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    "react-bootstrap": "^1.0.0-beta.14",
+    "react-dom": "^16.11.0",
+    "sass-loader": "^8.0.0",
+    "style-loader": "^1.0.0",
+    "typescript": "^3.6.4",
+    "webpack": "^4.41.2",
+    "webpack-cli": "^3.3.9"
+  },
+  "scripts": {
+    "build": "yarn webpack --mode=production"
```
Comment (MichaelDorkhom) [inline package.json]: Can you add `"prepare" : "npm run build"` here. See https://itnext.io/step-by-step-building-and-publishing-an-npm-typescript-package-44fe7164964c

**Comment by MichaelDorkhom** [comment]
@julianjuko Can you remove all the dist files. The `prepare` script will automatically run build when a client installs the package from npm

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,16 @@
+.DS_Store
```
Comment (MichaelDorkhom) [inline .gitignore]: Can you add the `dist` folder to gitignore

File: src/AdminLogo/index.scss [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+  h1 {
+    font-size: 30px;
+    font-family: 'LatoBold', 'Lucida Grande', 'Lucida Sans Unicode', Helvetica, sans-serif;
+    letter-spacing: -1.5px;
+    color: #1d2939;
+    line-height: 30px;
+    margin: 0;
+  }
+}
```
Comment (MichaelDorkhom) [inline src/AdminLogo/index.scss]: Can you make sure you add an empty to EOF

File: src/Footer/index.scss [hunk 6]
```diff
@@ -0,0 +1,15 @@
+@import '../style/colors';
+
+#footer {
+  padding: 0 0 0 200px;
+  color: #dadada;
```
Comment (MichaelDorkhom) [inline src/Footer/index.scss]: These colors should be constants

File: src/Navbar/Navbar.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+import './index.scss'
+
+interface IProps {
+  logo: JSX.Element | null
+}
+
+const NavbarC: React.FC<IProps> = ({ logo }): JSX.Element => {
+  return (
+    <>
```
Comment (MichaelDorkhom) [inline src/Navbar/Navbar.tsx]: Don't need fragment here

File: src/SideNav/SideNav.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import { INavLink } from '..'
+
+import './index.scss'
+
+interface IProps {
+  navLinks: INavLink[]
+  handleNavTo: (route: string) => void
+}
+
+
```
Comment (MichaelDorkhom) [inline src/SideNav/SideNav.tsx]: Double empty line

File: src/SideNav/SideNav.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+const SideNav: React.FC<IProps> = ({ navLinks, handleNavTo }) => {
+
+  const renderNavLink = (link: INavLink): JSX.Element => {
```
Comment (MichaelDorkhom) [inline src/SideNav/SideNav.tsx]: Can you move this to another file

File: src/SideNav/components/SideNavLink/SideNavLink.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  faIcon: IconProp
+}
+
+const SideNavLink: React.FC<IProps> = ({
+  handleNavigateTo,
+  label,
+  route,
+  faIcon
+}) => {
+
```
Comment (MichaelDorkhom) [inline src/SideNav/components/SideNavLink/SideNavLink.tsx]: No need for a blank line here

File: src/SideNav/components/SideNavLink/index.scss [hunk 11]
```diff
@@ -0,0 +1,21 @@
+@import '../../../style/colors';
+
+.navbar-nav {
+  opacity: 0.6;
+  padding: 5px;
+  cursor: pointer;
+  border-radius: 4px;
+
+  &:hover {
+    background: #f6f7fa;
```
Comment (MichaelDorkhom) [inline src/SideNav/components/SideNavLink/index.scss]: Make sure all colors are constants

File: src/style/colors.scss [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+$teal: rgb(0, 171, 176);
+$darkteal: rgb(4, 156, 165);
+$lightgrey: rgb(222, 226, 230);
+$darkgrey: rgb(107, 107, 107);
+$darkblue: rgb(0, 21, 61) !important;
+$ade: rgb(226, 175, 75);
+$subtherapeutic: rgb(119, 126, 167);
+$see: rgb(247, 135, 100);
+$inRange: rgb(140, 168, 170);
+$warning: rgb(255, 193, 7) !important;
```
Comment (MichaelDorkhom) [inline src/style/colors.scss]: Shouldn't need to use `!important` here

File: src/index.ts (hunk 1 above)
Comment (julianjuko) [reply src/index.ts]: it's actually more relevant to `SideNav`, but I guess putting it there makes sense

File: src/style/colors.ts (hunk 2 above)
Comment (julianjuko) [reply src/style/colors.ts]: I think I'll leave them here for when we start making Cohere use this lib

File: package.json (hunk 3 above)
Comment (julianjuko) [reply package.json]: pretty cool, maybe we can add some other npm scripts as well down the track like `preversion` etc

File: src/AdminLogo/index.scss (hunk 5 above)
Comment (julianjuko) [reply src/AdminLogo/index.scss]: ah didn't realise this applies to sass too, will do

**Pull request merged**